app/src/main/java/ca/llamabagel/transpo/search/data/SearchRepository.kt

```diff
@@ -55,53 +55,53 @@ class SearchRepository @Inject constructor(
     val recentFlow get() = recentChannel.asFlow()
 
     suspend fun getSearchResults(query: String, filters: SearchFilter) {
-        getRoutes(query.takeIf { filters.routes }.orEmpty())
-        getStops(query.takeIf { filters.stops }.orEmpty())
-        getPlaces(query.takeIf { filters.places }.orEmpty())
-        getRecent(query.takeIf { filters.recent }.orEmpty(), filters)
+        val recentList = if (query.isEmpty()) {
```

Comment:
This could be streamlined to combine duplicated code

```kotlin
val recentList = if (query.isEmpty()) {
   database.recentSearchQueries.getMostRecent(...)
} else {
  database.recentSearchQueries.searchRecentStops(...)
}
  .executeAsList()
  .map { ... }
```

Not with this formatting, of course...